### PR TITLE
Move logging to fern

### LINKED ISSUE #285: Move logging to fern
`fern` is able to log to stdout as well as system logs by attaching handlers during the initialisation. This would move our logging dependency to one crate instead of two and would remove the awkward logging code in `main.rs`. Yes, `fern` does use `syslog` under the hood but it would clean up the `setup_logger` function to only use crate calls from `fern` instead of two different crates.

### PR DESCRIPTION
**Pull request by mh84**
Closes #285

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -21,42 +20,55 @@ mod process;
...
+    logger.apply().expect("Couldn't initialize logger");
 }
 
 fn main() {
     let mut cli_config: CliConfig = CliConfig::from_args();
 
-    let is_daemon = !cli_config.no_daemon;
-    let is_verbose = cli_config.verbose;
-    setup_logger(is_daemon, is_verbose);
+    let is_deamon = !cli_config.no_daemon;
```
Comment (SirWindfield) [inline src/main.rs]: It's actually called [daemon](https://en.wikipedia.org/wiki/Daemon_(computing)).

**Comment by mh84** [comment]
Oh sorry, i will fix that typo :)

**Comment by SirWindfield** [comment]
I'll merge it later on this day, after v0.2.20 released. Thanks for your contribution!

**Comment by SimonPersson** [comment]
Thanks, merged!

**Pull request merged**